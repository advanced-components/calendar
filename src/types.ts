export interface VEvent {
  title: string;
  start: number;
  end: number;
}
