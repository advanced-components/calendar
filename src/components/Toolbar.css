.cal-toolbar {
  width: 100%;
  border-bottom: 1px solid #eee;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
}
