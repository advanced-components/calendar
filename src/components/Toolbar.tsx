import React from 'react';
import { Views, Navigate } from '../types';
import './Toolbar.css';

export type ToolbarProps = {
  onNavigate: (to: Navigate) => void;
  onChangeView: (view: Views) => void;
};

export const Toolbar = (props: ToolbarProps) => (
  <div className='cal-toolbar'>
    <div className='cal-toolbar__left'>
      <div className='button-group'>
        <div
          className='button cal-toolbar__button-today'
          onClick={() => props.onNavigate(Navigate.TODAY)}
        >
          Today
        </div>
        <div
          className='button cal-toolbar__button-previous'
          onClick={() => props.onNavigate(Navigate.PREVIOUS)}
        >
          {'<'}
        </div>
        <div
          className='button cal-toolbar__button-next'
          onClick={() => props.onNavigate(Navigate.NEXT)}
        >
          {'>'}
        </div>
      </div>
    </div>
    <div className='cal-toolbar__right'>
      <div className='button-group'>
        <div
          className=' button cal-toolbar__button-day'
          onClick={() => props.onChangeView(Views.DAY)}
        >
          Day
        </div>
        <div
          className=' button cal-toolbar__button-week'
          onClick={() => props.onChangeView(Views.WEEK)}
        >
          Week
        </div>
        <div
          className=' button cal-toolbar__button-month'
          onClick={() => props.onChangeView(Views.MONTH)}
        >
          Month
        </div>
      </div>
    </div>
  </div>
);
