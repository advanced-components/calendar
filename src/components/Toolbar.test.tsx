import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import { Views, Navigate } from '../types';

import { Toolbar, ToolbarProps } from './Toolbar';

const onNavigate = jest.fn();
const onChangeView = jest.fn();

const defaultProps: ToolbarProps = {
  onNavigate: onNavigate,
  onChangeView: onChangeView
};

describe('<Toolbar />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Toolbar {...defaultProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('call onNavigate(TODAY)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-today').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(Navigate.TODAY);
  });

  it('call onNavigate(NEXT)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-next').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(Navigate.NEXT);
  });

  it('call onNavigate(PREVIOUS)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-previous').simulate('click');
    expect(onNavigate).toHaveBeenCalledWith(Navigate.PREVIOUS);
  });

  it('call onChangeView(DAY_VIEW)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-day').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(Views.DAY);
  });

  it('call onChangeView(WEEK_VIEW)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-week').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(Views.WEEK);
  });

  it('call onChangeView(MONTH_VIEW)', () => {
    const component = shallow(<Toolbar {...defaultProps} />);
    component.find('.cal-toolbar__button-month').simulate('click');
    expect(onChangeView).toHaveBeenCalledWith(Views.MONTH);
  });
});
