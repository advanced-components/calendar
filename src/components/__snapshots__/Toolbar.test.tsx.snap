// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> renders correctly 1`] = `
<div
  className="cal-toolbar"
>
  <div
    className="cal-toolbar__left"
  >
    <div
      className="button-group"
    >
      <div
        className="button cal-toolbar__button-today"
        onClick={[Function]}
      >
        Today
      </div>
      <div
        className="button cal-toolbar__button-previous"
        onClick={[Function]}
      >
        &lt;
      </div>
      <div
        className="button cal-toolbar__button-next"
        onClick={[Function]}
      >
        &gt;
      </div>
    </div>
  </div>
  <div
    className="cal-toolbar__right"
  >
    <div
      className="button-group"
    >
      <div
        className=" button cal-toolbar__button-day"
        onClick={[Function]}
      >
        Day
      </div>
      <div
        className=" button cal-toolbar__button-week"
        onClick={[Function]}
      >
        Week
      </div>
      <div
        className=" button cal-toolbar__button-month"
        onClick={[Function]}
      >
        Month
      </div>
    </div>
  </div>
</div>
`;
